tinyagentos/projects/task_store.py

```
@@ -698,3 +722,107 @@ async def list_comments(self, task_id: str) -> list[dict]:
+        if not candidates:
+            return await self.get_checklist_item(item_id)
+        sets: list[str] = []
+        params: list = []
+        for col, val in candidates:
+            sets.append(f"{col} = ?")
+            params.append(val)
+        sets.append("updated_at = ?")
+        params.append(now)
+        params.append(item_id)
+        await self._db.execute(
+            f"UPDATE task_checklist_items SET {', '.join(sets)} WHERE id = ?", params
```

Ruff (0.16.1)

[error] Possible SQL injection vector through string-based query construction

(S608)